#Association between Device Type and Customer Satisfaction

**Background:**

Mizzare Corporation has collected data on customer satisfaction levels for two types of smart home devices: Smart Thermostats and Smart Lights. They want to determine if there's a significant association between the type of device purchased and the customer's satisfaction level.

**Data Provided:**

The data is summarized in a contingency table showing the counts of customers in each satisfaction level for both types of devices:


Satisfaction	: Smart Thermostat	Smart Light	Total

Very Satisfied	: 50	70	120

Satisfied :	80	100	180

Neutral	 :60	90	150

Unsatisfied	:30	50	80

Very Unsatisfied	:20	50	70

Total	:240	360	600

**Objective:**

To use the Chi-Square test for independence to determine if there's a significant association between the type of smart home device purchased (Smart Thermostats vs. Smart Lights) and the customer satisfaction level.








In [ ]:
import pandas as pd #importing required liqaries
import numpy as np
import scipy.stats as stats
from scipy.stats import chi2_contingency

In [ ]:
data = pd.DataFrame({'Smart Thermostat': [50, 80, 60, 30, 20],'Smart Light': [70, 100, 90, 50, 50]}, index=['Very Satisfied', 'Satisfied', 'Neutral', 'Unsatisfied', 'Very Unsatisfied']) #Creatinng a dataframe from provided data.
data

Smart Thermostat  Smart Light
Very Satisfied                  50           70
Satisfied                       80          100
Neutral                         60           90
Unsatisfied                     30           50
Very Unsatisfied                20           50

#Define Hypothesis:
 **Null Hypothesis:** There is no significant association between the type of smart home device(Smart Thermostat vs. Smart Light) and customer satisfaction.

 **Alternative Hypothesis:**  There is  significant association between the type of smart home device(Smart Thermostat vs. Smart Light) and customer satisfaction.



#Chi-Square Statistic:

In [ ]:
stats.chi2_contingency(data) #Performing Chi-Square test

Chi2ContingencyResult(statistic=5.638227513227513, pvalue=0.22784371130697179, dof=4, expected_freq=array([[ 48.,  72.],
       [ 72., 108.],
       [ 60.,  90.],
       [ 32.,  48.],
       [ 28.,  42.]]))

#Critical Value:


In [ ]:
dof = 4 # Degrees of freedom (df)
alpha = 0.05 # Significance level (alpha)
critical_value = stats.chi2.ppf(1 - alpha, dof) # Calculating the critical value
critical_value

9.487729036781154

#Decision:

Here Chi-Square statistic is 5.64, P-value is 0.23 & critical value is 9.49.

Since the critical-value is greater than chi-square statistic value,we shall counclude that we faill to reject null hypothesis, that there is no significant association between the type of smart home device(Smart Thermostat vs. Smart Light) and customer satisfaction.

#Conclusion:

We failed to reject the null hypothesis, indicating that there is no significant association between the device type and customer satisfaction levels. This suggests that satisfaction levels do not significantly vary depending on whether customers purchased a Smart Thermostat or a Smart Light.